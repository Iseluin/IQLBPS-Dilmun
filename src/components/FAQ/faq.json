{
  "questions": [
    {
      "question": "What is the return policy?",
      "answer": "There must be a deal between the customers and the sellers to determine how returning items should be and our staff is not responsible of it."
    },
    {
      "question": "Can I cancel my order?",
      "answer": "Sure! you can cancel your order within first 12 hours."
    },
    {
      "question": "What methods of payment you take?",
      "answer": "You can pay online via credit card or pay at the door while dealing with the seller."
    },
    {
      "question": "How do I track my order?",
      "answer": "You can have the phone number of the company which does the delivering and ask them where your order is."
    },
    {
      "question": "What is the policy for late/non-delivery of items ordered online?",
      "answer": "Our customer service team will be there alwyas to resolve your issue, you can contact us."
    }
  ]
}