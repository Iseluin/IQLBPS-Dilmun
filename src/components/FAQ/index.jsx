import { Disclosure, Transition } from '@headlessui/react';
import data from './faq.json';

function FAQ() {
  return (
    <section className="bg-background py-10 mx-auto">
      <h2 className="text-4xl mb-10 text-primary font-bold">
        Frequently Asked Questions
      </h2>
      <div className=" mx-14  px-4 flex flex-col lg:flex-row gap-4 justify-between items-center">
        <div className="w-full lg:w-1/2 max-w-md p-2">
          {data.questions.map((item, index) => (
            <Disclosure key={item.question}>
              {({ open }) => (
                <>
                  <Disclosure.Button
                    className={`flex w-full justify-between border-b-2 border-primary px-4 py-4 text-left font-medium hover:text-white hover:bg-primary focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-opacity-75 text-md ${
                      open
                        ? ' bg-primary text-white '
                        : ' bg-background  text-primary '
                    } ${index === 0 && ' rounded-t-lg '} ${
                      index === data.length - 1 &&
                      ' rounded-b-lg border-b-none '
                    }`}
                  >
                    <span>{item.question}</span>
                    <svg
                      xmlns="http://www.w3.org/2000/svg"
                      fill="none"
                      viewBox="0 0 24 24"
                      strokeWidth="1.5"
                      stroke="currentColor"
                      className={`${
                        open ? 'rotate-180 transform' : ''
                      } h-5 w-5`}
                    >
                      <path
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        d="M4.5 15.75l7.5-7.5 7.5 7.5"
                      />
                    </svg>
                  </Disclosure.Button>
                  <Transition
                    show={open}
                    enter="transition duration-100 ease-out"
                    enterFrom="transform scale-95 opacity-0"
                    enterTo="transform scale-100 opacity-100"
                    leave="transition duration-75 ease-out"
                    leaveFrom="transform scale-100 opacity-100"
                    leaveTo="transform scale-95 opacity-0"
                  >
                    <Disclosure.Panel
                      static
                      className="px-4 py-4 text-md text-gray-500 text-left"
                    >
                      {item.Answer}
                    </Disclosure.Panel>
                  </Transition>
                </>
              )}
            </Disclosure>
          ))}
        </div>
        <div className="w-1/2 h-full">
          <img
            className="w-full h-full max-h-96 object-contain"
            src="/assets/img/keila-hotzel-lFmuWU0tv4M-unsplash.jpg"
            alt="fkndfjn"
          />
        </div>
      </div>
    </section>
  );
}

export default FAQ;
