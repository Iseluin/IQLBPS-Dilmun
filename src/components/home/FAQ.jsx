import React from 'react';

function FAQ() {
  return <div>FAQ Component</div>;
}

export default FAQ;
