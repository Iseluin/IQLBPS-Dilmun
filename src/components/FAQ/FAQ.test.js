import renderer from 'react-test-renderer';
import FAQ from '.';

test('FAQ renders correctly', () => {
  const tree = renderer.create(<FAQ />).toJSON();
  expect(tree).toMatchSnapshot();
});
