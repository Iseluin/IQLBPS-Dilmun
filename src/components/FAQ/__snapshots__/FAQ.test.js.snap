// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ renders correctly 1`] = `
<section
  className="bg-background py-20 mx-auto min-h-screen"
>
  <h2
    className="text-4xl mb-10 text-primary font-bold text-center"
  >
    Frequently Asked Questions
  </h2>
  <div
    className="max-w-7xl mx-auto px-4 py-10 lg:max-h-100 flex flex-col-reverse lg:flex-row gap-4 justify-between items-center"
  >
    <div
      className="w-full lg:w-1/2 max-w-lg p-2 "
    >
      <button
        aria-expanded={false}
        className="flex w-full justify-between border-b-2 border-primary px-4 py-4 text-left font-medium hover:text-white hover:bg-primary focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-opacity-75 text-md  bg-background text-primary   rounded-t-lg  false"
        data-headlessui-state=""
        id="headlessui-disclosure-button-:r0:"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="w-3/4 md:w-full"
        >
          What is the return policy?
        </span>
        <svg
          className="rotate-180 duration-200 transform h-5 w-5"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.5 15.75l7.5-7.5 7.5 7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
      <button
        aria-expanded={false}
        className="flex w-full justify-between border-b-2 border-primary px-4 py-4 text-left font-medium hover:text-white hover:bg-primary focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-opacity-75 text-md  bg-background text-primary  false false"
        data-headlessui-state=""
        id="headlessui-disclosure-button-:r2:"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="w-3/4 md:w-full"
        >
          Can I cancel my order?
        </span>
        <svg
          className="rotate-180 duration-200 transform h-5 w-5"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.5 15.75l7.5-7.5 7.5 7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
      <button
        aria-expanded={false}
        className="flex w-full justify-between border-b-2 border-primary px-4 py-4 text-left font-medium hover:text-white hover:bg-primary focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-opacity-75 text-md  bg-background text-primary  false false"
        data-headlessui-state=""
        id="headlessui-disclosure-button-:r4:"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="w-3/4 md:w-full"
        >
          What methods of payment you take?
        </span>
        <svg
          className="rotate-180 duration-200 transform h-5 w-5"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.5 15.75l7.5-7.5 7.5 7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
      <button
        aria-expanded={false}
        className="flex w-full justify-between border-b-2 border-primary px-4 py-4 text-left font-medium hover:text-white hover:bg-primary focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-opacity-75 text-md  bg-background text-primary  false false"
        data-headlessui-state=""
        id="headlessui-disclosure-button-:r6:"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="w-3/4 md:w-full"
        >
          How do I track my order?
        </span>
        <svg
          className="rotate-180 duration-200 transform h-5 w-5"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.5 15.75l7.5-7.5 7.5 7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
      <button
        aria-expanded={false}
        className="flex w-full justify-between border-b-2 border-primary px-4 py-4 text-left font-medium hover:text-white hover:bg-primary focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-opacity-75 text-md  bg-background text-primary  false false"
        data-headlessui-state=""
        id="headlessui-disclosure-button-:r8:"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="w-3/4 md:w-full"
        >
          What is the policy for late/non-delivery of items ordered online?
        </span>
        <svg
          className="rotate-180 duration-200 transform h-5 w-5"
          fill="none"
          stroke="currentColor"
          strokeWidth="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.5 15.75l7.5-7.5 7.5 7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </button>
    </div>
    <div
      className="w-full lg:w-1/2 h-full"
    >
      <img
        alt="frequently asked questions"
        className="w-full h-full max-h-96 object-contain 	"
        src="/assets/img/faq.png"
      />
    </div>
  </div>
</section>
`;
